python/ray/serve/_private/http_proxy.py

```diff
@@ -75,9 +78,21 @@ async def _send_request_to_handle(handle, scope, receive, send) -> str:
             assignment_task.cancel()
         try:
             object_ref = await assignment_task
-            result = await object_ref
-            client_disconnection_task.cancel()
-            break
+            _, timed_out = await asyncio.wait(
```

Comment:
cancel time out tasks.

Reply:
I canceled the `client_disconnect_task`. I'm not sure how to cancel the `object_ref` from `timed_out`. The `object_ref` is [a Ray object ref](https://github.com/ray-project/ray/blob/437cf14e737df679ccf71a0e38d52f37b0e436fa/python/ray/serve/_private/router.py#L174-L178), not an `asyncio.Task`.

I believe the only ways to terminate a Ray remote call are [`ray.kill`](https://docs.ray.io/en/releases-2.0.1/ray-core/package-ref.html#ray-kill) (which would kill the whole replica) or [`ray.cancel`](https://docs.ray.io/en/releases-2.0.1/ray-core/package-ref.html#ray-cancel) (which can't be used on Ray actor method calls). I don't think there's a straightforward way to kill the call. Do you have any suggestions on how to kill it?

Reply:
good call, i don't think we are able to do it. We can leave it as it is.